e2e/tests/ui/pages/Toolbar.ts

```diff
@@ -1,81 +1,126 @@
 import { expect, type Locator, type Page } from "@playwright/test";
 
-export class Toolbar {
+type TFilterValue = "string" | "dateRange" | "multiSelect" | "typeahead";
+
+type TDateRange = { from: string; to: string };
+type TMultiValue = string[];
+
+type FilterValueTypeMap = {
+  string: string;
+  dateRange: TDateRange;
+  multiSelect: TMultiValue;
+  typeahead: TMultiValue;
+};
+
+type FilterValueType<TFilter extends Record<string, TFilterValue>> = {
+  [K in keyof TFilter]: FilterValueTypeMap[TFilter[K]];
+};
+
+function isStringFilter<K extends string, T extends Record<K, TFilterValue>>(
+  type: T[K],
+  value: unknown,
+): value is string {
+  return type === "string";
+}
+
+function isDateRangeFilter<K extends string, T extends Record<K, TFilterValue>>(
+  type: T[K],
+  value: unknown,
+): value is TDateRange {
+  return type === "dateRange";
+}
+
+function isMultiSelectFilter<
+  K extends string,
+  T extends Record<K, TFilterValue>,
+>(type: T[K], value: unknown): value is TMultiValue {
+  return type === "multiSelect";
+}
+
+function isTypeaheadFilter<K extends string, T extends Record<K, TFilterValue>>(
+  type: T[K],
+  value: unknown,
+): value is TMultiValue {
+  return type === "typeahead";
+}
+
```

Comment:
`value` on these functions were never used.
I see the purpose this code block is to return the type of the filtering condition. As we retrieve filtering type on line [81](https://github.com/guacsec/trustify-ui/pull/822/files#diff-b9bdf33fd65e9cb221148c89f51820bb675144a96588f2cfb79864ed4b5bfb6fR81), this can be simply used with if condition without these functions 
```
if (filterType === "string") {
  await this.applyTextFilter(filterName as keyof TFilter, filterValue as string);
}
```
instead of
```
if (isStringFilter(filterType, filterValue)) {
        await this.applyTextFilter(filterName, filterValue);
}
```